From e668d935dd720ef6166266d51b0503bfd1d3a272 Mon Sep 17 00:00:00 2001
From: Tomas Glozar <tglozar@gmail.com>
Date: Tue, 18 Nov 2025 21:19:34 +0100
Subject: [PATCH 94/96] ia64: Guard code requiring CONFIG_CRASH_DUMP

Both crash_notes and parse_crashconfig() depend on CONFIG_CRASH_DUMP.

Put crash_save_this_cpu() and setup_crashkernel() behind #ifdef
CONFIG_CRASH_DUMP to prevent build failure on CONFIG_EXEC &&
!CONFIG_CRASH_DUMP.

Signed-off-by: Tomas Glozar <tglozar@gmail.com>
---
 arch/ia64/kernel/crash.c | 7 +++++++
 arch/ia64/kernel/setup.c | 2 +-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/arch/ia64/kernel/crash.c b/arch/ia64/kernel/crash.c
index 1d46d8acac1b..a77654a73bfb 100644
--- a/arch/ia64/kernel/crash.c
+++ b/arch/ia64/kernel/crash.c
@@ -31,6 +31,7 @@ static int kdump_on_fatal_mca = 1;
 
 extern void ia64_dump_cpu_regs(void *);
 
+#ifdef CONFIG_CRASH_DUMP
 static DEFINE_PER_CPU(struct elf_prstatus, elf_prstatus);
 
 void
@@ -60,6 +61,12 @@ crash_save_this_cpu(void)
 			sizeof(*prstatus));
 	final_note(buf);
 }
+#else
+void
+crash_save_this_cpu(void)
+{
+}
+#endif
 
 #ifdef CONFIG_SMP
 static int
diff --git a/arch/ia64/kernel/setup.c b/arch/ia64/kernel/setup.c
index f43bedab6649..3dbd37da666a 100644
--- a/arch/ia64/kernel/setup.c
+++ b/arch/ia64/kernel/setup.c
@@ -252,7 +252,7 @@ static int __init register_memory(void)
 __initcall(register_memory);
 
 
-#ifdef CONFIG_KEXEC
+#if defined(CONFIG_KEXEC) && defined(CONFIG_CRASH_DUMP)
 
 /*
  * This function checks if the reserved crashkernel is allowed on the specific
-- 
2.25.1

